### DISCUSSION TITLE
MicroPython-powered respiratory coach for clinical research

### DISCUSSION THREAD

**Discussion by JDRBoston** [micropython/micropython#12288] [Show and tell]
Here's how we used MicroPython to control a simple device that enhances our clinical research. We are interested in developing new methods to monitor the breathing of our patients, particularly if they use algorithmic rather than hardware approaches, leverage monitors that are already being used for the patient, and might later be scaled for mobile patient monitoring environments. Accordingly, we are developing an algorithm that converts surface ECG voltage wave forms, already obtained using inexpensive and portable heart monitors, to respiratory rate. For this work, we needed to obtain ECG data while subjects breathed at specific respiratory rates and patterns. But, we found no reliable and customizable way to coach the subjects. During the development of patient ventilator systems, some have used audio recordings or metronomes, or even some 'mindful breathing' apps, to do this. But, they all had limited capabilities that did not meet our needs. So we used an ESP32 powered by MicroPython and a few external components to make a flexible, small, battery powered device, which we call 'RespiCo', that produces prescribed respiratory coaching outputs in audio, visible, and tactile forms for our research purposes. A research paper describing the build, software, and validations can be accessed at HardwareX, linked below. It meets our needs. But others might like to build and use it for mindful breathing work, to build-up their respiratory muscles to treat certain lung diseases, or to enhance their athleticism. Please let me know what you think. Thank you, J

https://www.hardware-x.com/article/S2468-0672(22)00080-3/fulltext

**Comment by shariltumin** [comment]
Hey, thanks for sharing. Interesting post. I have enough info to try to make my own breathing trainer. I do have one question though. What is the best duration of a practice session. Does it depend on health conditions, age and a like? I am interested in such a device for the formation of a good breathing habit. Any suggestion.

**Reply by JDRBoston** [reply]
Hi shariltumin, thank you for your interest! You are right that the underlying health conditions, age, etc. modify the best duration of a practice session. Please send me an email at the address mapped to my name in the article and I'll be happy to share with you what I know about respiratory muscle conditioning more deeply. Best wishes, J
